src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -991,13 +998,22 @@ protected long getApproxSizeOfRecordInBytes(SinkRecord kafkaSinkRecord) {
     for (Map.Entry<String, Object> entry : tableRow.entrySet()) {
       sinkRecordBufferSizeInBytes += entry.getKey().length() * 2L;
       // Can Typecast into string because value is JSON
-      sinkRecordBufferSizeInBytes += ((String) entry.getValue()).length() * 2L; // 1 char = 2 bytes
+      Object value = entry.getValue();
+      if (value instanceof String) {
+        sinkRecordBufferSizeInBytes += ((String) value).length() * 2L; // 1 char = 2 bytes
+      } else {
+        // for now it could only be a list of string
+        for (String s : (List<String>) value) {
+          sinkRecordBufferSizeInBytes += s.length() * 2L;
+        }
+      }
```

Comment:
do you need this now? We can add it once we support list? Add a TODO JIRA here if needed.

Reply:
Is it ok if I leave the behavior correct on the connector side?